### ISSUE TITLE
ccache reports no input file found for rsp with single input file

### ISSUE THREAD

**Issue by jbrownsw** [ccache/ccache#1308]
### How to reproduce ###
See the following logs which I've replaced most identifying information, but kept all the options to try to keep it as consistent with the original as possible for reproducibility.

ccache log:
```
[2023-07-05T09:17:23.782407 43172] === CCACHE 4.8.2 STARTED =========================================
[2023-07-05T09:17:23.782552 43172] Configuration file: C:\Users\user\AppData\Local\ccache\ccache.conf
[2023-07-05T09:17:23.782577 43172] System configuration file: C:\ProgramData\ccache\ccache.conf
[2023-07-05T09:17:23.782577 43172] Config: (default) absolute_paths_in_stderr = false
[2023-07-05T09:17:23.782577 43172] Config: (default) base_dir = 
[2023-07-05T09:17:23.782577 43172] Config: (default) cache_dir = C:\Users\user\AppData\Local\ccache
[2023-07-05T09:17:23.782577 43172] Config: (default) compiler = 
[2023-07-05T09:17:23.782577 43172] Config: (default) compiler_check = mtime
[2023-07-05T09:17:23.782577 43172] Config: (default) compiler_type = auto
[2023-07-05T09:17:23.782577 43172] Config: (default) compression = true
[2023-07-05T09:17:23.782577 43172] Config: (default) compression_level = 0
[2023-07-05T09:17:23.782577 43172] Config: (default) cpp_extension = 
[2023-07-05T09:17:23.782577 43172] Config: (default) debug = false
[2023-07-05T09:17:23.782577 43172] Config: (default) debug_dir = 
[2023-07-05T09:17:23.782577 43172] Config: (default) depend_mode = false
[2023-07-05T09:17:23.782577 43172] Config: (default) direct_mode = true
[2023-07-05T09:17:23.782577 43172] Config: (default) disable = false
[2023-07-05T09:17:23.782577 43172] Config: (default) extra_files_to_hash = 
[2023-07-05T09:17:23.782577 43172] Config: (default) file_clone = false
[2023-07-05T09:17:23.782577 43172] Config: (default) hard_link = false
[2023-07-05T09:17:23.782577 43172] Config: (default) hash_dir = true
[2023-07-05T09:17:23.782577 43172] Config: (default) ignore_headers_in_manifest = 
[2023-07-05T09:17:23.782577 43172] Config: (default) ignore_options = 
[2023-07-05T09:17:23.782577 43172] Config: (default) inode_cache = true
[2023-07-05T09:17:23.782577 43172] Config: (default) keep_comments_cpp = false
[2023-07-05T09:17:23.782577 43172] Config: (environment) log_file = C:/work/project/ccache.log
[2023-07-05T09:17:23.782577 43172] Config: (default) max_files = 0
[2023-07-05T09:17:23.782577 43172] Config: (default) max_size = 5.0 GiB
[2023-07-05T09:17:23.782577 43172] Config: (default) msvc_dep_prefix = Note: including file:
[2023-07-05T09:17:23.782577 43172] Config: (default) namespace = 
[2023-07-05T09:17:23.782577 43172] Config: (default) path = 
[2023-07-05T09:17:23.782577 43172] Config: (default) pch_external_checksum = false
[2023-07-05T09:17:23.782577 43172] Config: (default) prefix_command = 
[2023-07-05T09:17:23.782577 43172] Config: (default) prefix_command_cpp = 
[2023-07-05T09:17:23.782577 43172] Config: (default) read_only = false
[2023-07-05T09:17:23.782577 43172] Config: (default) read_only_direct = false
[2023-07-05T09:17:23.782577 43172] Config: (default) recache = false
[2023-07-05T09:17:23.782577 43172] Config: (default) remote_only = false
[2023-07-05T09:17:23.782577 43172] Config: (default) remote_storage = 
[2023-07-05T09:17:23.782577 43172] Config: (default) reshare = false
[2023-07-05T09:17:23.782577 43172] Config: (default) run_second_cpp = true
[2023-07-05T09:17:23.782577 43172] Config: (default) sloppiness = 
[2023-07-05T09:17:23.782577 43172] Config: (default) stats = true
[2023-07-05T09:17:23.782577 43172] Config: (default) stats_log = 
[2023-07-05T09:17:23.782577 43172] Config: (default) temporary_dir = C:\Users\user\AppData\Local\ccache/tmp
[2023-07-05T09:17:23.782577 43172] Config: (default) umask = 
[2023-07-05T09:17:23.782665 43172] Command line: C:\work\project\build\cl.exe @C:\Users\user\AppData\Local\Temp\tmpdeaeb6b2b5774db5b2a05dc156afa3c6.rsp
[2023-07-05T09:17:23.790873 43172] Hostname: ADC-DTD9HK3
[2023-07-05T09:17:23.790909 43172] Working directory: C:/work/project/build/windows-x64-vc16-debug/coreproj/main/proj
[2023-07-05T09:17:23.791331 43172] Compiler: C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64/cl.exe
[2023-07-05T09:17:23.791351 43172] Compiler type: msvc
[2023-07-05T09:17:23.791823 43172] No input file found
[2023-07-05T09:17:23.791852 43172] Failed; falling back to running the real compiler
[2023-07-05T09:17:23.791866 43172] Executing "C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64/cl.exe" @C:\Users\user\AppData\Local\Temp\tmpdeaeb6b2b5774db5b2a05dc156afa3c6.rsp
[2023-07-05T09:17:23.791884 43172] Result: no_input_file
```

rsp file contents clearly has a single input file:
```
/c /IC:\work\project\coreproj\main\proj\include /I"C:\Users\user\.conan\data\ffmpeg\5.1.2-b\company\testing\package\584b545bf9ed632dc1501f13313093e81d60449e\include" /IC:\Users\user\.conan\data\ftd2xx_windows\2.12.28\company\testing\package\d057732059ea44a47760900cb5e4855d2bea8714\. /Z7 /nologo /W3 /WX /diagnostics:column /Od /Ob0 /D _MBCS /D WIN32 /D _WINDOWS /D NOMINMAX /D _WIN32_WINNT=0x0600 /D WINVER=0x0600 /D __STDC_LIMIT_MACROS /D _CRT_SECURE_NO_WARNINGS /D _SCL_SECURE_NO_WARNINGS /D _WINSOCK_DEPRECATED_NO_WARNINGS /D _ITERATOR_DEBUG_ARRAY_OVERLOADS /D ANALYTICS_LIB /D COREPROJ_OPENSSL /D COREPROJ_FFMPEG /D COREPROJ_ZLIB /D ZLIB_CONST /D COREPROJ_LMDB /D COREPROJ_LIBARCHIVE /D COREPROJ_OPENCV /D COREPROJ_OPENH264 /D COREPROJ_STARTECH /D COREPROJ_INTELMEDIA /D BOOST_STACKTRACE_USE_NOOP /D BOOST_STACKTRACE_USE_WINDBG /D BOOST_STACKTRACE_USE_WINDBG_CACHED /D BOOST_ALL_NO_LIB /D LIBARCHIVE_STATIC /D "CMAKE_INTDIR=\"Debug\"" /Gm- /EHsc /RTC1 /MDd /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /GR /std:c++14 /Fo"proj.dir\Debug\\" /Fd"C:\work\project\build\windows-x64-vc16-debug\lib\proj.pdb" /external:W0 /Gd /TP /wd4355 /wd4018 /wd4099 /wd4244 /wd4101 /wd4996 /wd4267 /errorReport:queue  /external:I "C:/.conan/8a938d/1/include" /external:I "C:/Users/user/.conan/data/rapidjson/2020-04-01/company/testing/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include" /external:I "C:/Users/user/.conan/data/openssl/3.1.1/_/_/package/a6deb10c65388e171398bba44b51776561f03e02/include" /external:I "C:/Users/user/.conan/data/zlib/1.2.12/_/_/package/d057732059ea44a47760900cb5e4855d2bea8714/include" /external:I "C:/Users/user/.conan/data/libarchive/3.6.2/_/_/package/ad7a92d06afd092415583c25717e7a9f7bfc68a8/include" /external:I "C:/Users/user/.conan/data/lmdb/0.9.24/company/testing/package/d057732059ea44a47760900cb5e4855d2bea8714/include" /external:I "C:/Users/user/.conan/data/bzip2/1.0.8/_/_/package/589a23dff5fdb23a7fb851223eb766480ead0a9a/include" /w35038 "C:\work\project\coreproj\main\proj\src\path\to\file.cpp"
```

### Actual behavior ###
No input file found resulting in uncacheable call

### Expected behavior ###
Input file should be found so the call can be cached

### Environment ###
ccache version 4.8.2
Visual studio 2022 using 2019 build tools compiler

### Additional Details ###
This is an existing large cmake project I'm trying to add ccache to support to in order to help with build times. I think I have all the settings configured properly based on the docs. Here's some snippets of my cmake settings:
```
cmake_policy(SET CMP0141 NEW)
project(apex)

if (MSVC)
    file(COPY_FILE ${CCACHE_PROGRAM} ${CMAKE_BINARY_DIR}/cl.exe ONLY_IF_DIFFERENT)
    set(CMAKE_VS_GLOBALS
        "CLToolExe=cl.exe"
        "CLToolPath=${CMAKE_BINARY_DIR}"
        "TrackFileAccess=false"
        "UseMultiToolTask=true"
        "DebugInformationFormat=OldStyle"
    )
    set(CMAKE_MSVC_DEBUG_INFORMATION_FORMAT "$<$<CONFIG:Debug,RelWithDebInfo>:Embedded>")
else()
    set(CMAKE_CXX_COMPILER_LAUNCHER "${CCACHE_PROGRAM}")
    set(CMAKE_C_COMPILER_LAUNCHER "${CCACHE_PROGRAM}")
endif()
```

That last `CMAKE_MSVC_DEBUG_INFORMATION_FORMAT ` part wasn't in the docs, but I think it's something they broke/changed when they introduced the `CMP0141` policy because `DebugInformationFormat` was in my globals set to OldStyle, but the individual configuration was still ProgramDatabase (/Zi) until I added that.

**Comment by jrosdahl** [comment]
Thanks for the bug report.

This was fixed by #1303 (not yet part of a released ccache version).

**Comment by jrosdahl** [comment]
Duplicate of #1247
